src/nyx/scan_fastloader_way.cpp

```diff
@@ -252,15 +273,35 @@ namespace Nyxus
 				return 1;
 			}
 
+		#ifdef USE_ARROW
+			if (arrow_output) {
+
+				auto features = Nyxus::get_feature_values();
+
+				auto status = theEnvironment.arrow_writer->write(Nyxus::get_feature_values());
+				
+				if (!status.ok()) {
+                    // Handle read error
+                    auto err = status.ToString();
+                    throw std::runtime_error("Error writing Arrow file: " + err);
```

Comment:
Should we `throw`? This can be performance degrading if we are in a situation where we keep throwing. Are we catching this in the caller code? If not, may be just ignore/log and move on? 
Since this function already return error codes if anything goes wrong, we should just use the same error reporting mechanism instead of `throw`.